sklearn/cluster/tests/test_dbscan.py

```diff
@@ -224,21 +223,16 @@ def test_input_validation():
 
 def test_dbscan_badargs():
     # Test bad argument values: these should all raise ValueErrors
-    assert_raises(ValueError,
-                  dbscan,
-                  X, eps=-1.0)
-    assert_raises(ValueError,
-                  dbscan,
-                  X, algorithm='blah')
-    assert_raises(ValueError,
-                  dbscan,
-                  X, metric='blah')
-    assert_raises(ValueError,
-                  dbscan,
-                  X, leaf_size=-1)
-    assert_raises(ValueError,
-                  dbscan,
-                  X, p=-1)
+    with pytest.raises(ValueError):
```

Comment:
Could you parametrize this one:

```python
@pytest.mark.parametrize(
    "args",
    [{'eps': -1.0}, {'algorithm': 'blah'}, {'metric': 'blah'},
     {'lead_size': -1}, {'p': -1}]
)
def test_dbscan_bad_args(args):
    with pytest.raises(ValueError):
        dbscan(X, **args)
```

Reply:
@glemaitre Will do! :+1:

Reply:
> Could you parametrize this one:
> 
> ```python
> @pytest.mark.parametrize(
>     "args",
>     [{'eps': -1.0}, {'algorithm': 'blah'}, {'metric': 'blah'},
>      {'lead_size': -1}, {'p': -1}]
> )
> def test_dbscan_bad_args(args):
>     with pytest.raises(ValueError):
>         dbscan(X, **args)
> ```

Will this run as 
```
def test_dbscan_bad_args(args):
    with pytest.raises(ValueError):
        dbscan(X,  eps=-1.0, algorithm=blah, metric=blah, lead_size= -1, p=-1)
```
or 5 times with only individual argument like 
```
def test_dbscan_bad_args(args):
    with pytest.raises(ValueError):
         dbscan(X,  eps=-1.0)
```
```
def test_dbscan_bad_args(args):
    with pytest.raises(ValueError):
        dbscan(X,  algorithm=blah)
```

I couldn't find the relevant clear documentation for this at pytest docs